### DISCUSSION TITLE
Hybrid search

### DISCUSSION THREAD

**Discussion by GluttonousCat** [milvus-io/milvus#35542] [Q&A and General discussion]
Regarding the combination of coarse and fine ranking, I create two fields: one to store sparse vectors and another to store dense vectors, then use hybrid retrieval. My plan is: **1) Use the hybrid retrieval function. 2) Conduct two queries.** Does anyone know about the feasibility and performance of this approach?

**Comment by yhmo** [comment]
A tutorial demonstrates how to conduct a hybrid search with milvus: https://milvus.io/docs/hybrid_search_with_milvus.md#Hybrid-Search-with-Milvus

In the tutorial, the hybrid search internally does the following things:
1. Conduct two queries to the two vector fields, each query generates a topk result.
2. Use a reranker to rerank the two results to the final topk result.

The performance is mainly depend on the two queries.
Currently, it only supports weight and RRF rerankers. If you have a special requirement for the rerank logic, you can conduct two queries and do the reranking by yourself.
